drivers/eeprom/eeprom_stm32.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019 Kwon Tae-young <tykwon@m2i.co.kr>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/eeprom.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_EEPROM_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(eeprom_stm32);
+
+K_MUTEX_DEFINE(lock);
+
+struct eeprom_stm32_config {
+	u32_t addr;
+	size_t size;
+};
+
+static int eeprom_stm32_read(struct device *dev, off_t offset, void *buf,
+				size_t len)
+{
+	const struct eeprom_stm32_config *config = dev->config->config_info;
+	u8_t *pbuf = buf;
+
+	if (!len) {
+		return 0;
+	}
+
+	if ((offset + len) > config->size) {
+		LOG_WRN("attempt to read past device boundary");
+		return -EINVAL;
+	}
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	while (len) {
+		*pbuf = *(__IO uint32_t*)(config->addr + offset);
```

Comment:
Why are you casting to `uint32_t` pointer? Shouldn't it be `u8_t` instead?

```
*pbuf = *(__IO u8_t*)(config->addr + offset);
```

Reply:
Done